index.d.ts

```diff
@@ -1231,6 +1231,8 @@ declare module 'mongoose' {
   type PreSaveMiddlewareFunction<T> = (this: T, next: (err?: CallbackError) => void, opts: SaveOptions) => void | Promise<void>;
   type PostMiddlewareFunction<ThisType, ResType = any> = (this: ThisType, res: ResType, next: (err?: CallbackError) => void) => void | Promise<void>;
   type ErrorHandlingMiddlewareFunction<ThisType, ResType = any> = (this: ThisType, err: NativeError, res: ResType, next: (err?: CallbackError) => void) => void;
+  
+  type Index = { [key: string]: boolean | number | string | Index }
```

Comment:
Why is this `| Index` here? I don't know of any recursive index definitions.

Reply:
My bad, changed it to mongodb's index type, It's my first ever pr on a project like this. it is not much but it should help with intellisense.